co2calculator/degree_days.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
```

Comment:
A great approach!
However, there is still considerable work to do to implement this into our code:
- Is this still part of the scope of the emission calculator?
- How should the degree days be integrated into the heating emissions calculation?
- We don't even have the temporal component in the emission calculator.

As this MR is basically just the script degree_days.py, it will be easy to rebase this branch in main any time whenever someone wants to work on it again. Thus, we don't need to merge it now.